RavenDB-17812 Making sure we wait a bit before getting index performance. Otherwise we will have a race in some of the executions.

omerlewitz opened the pull request:
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17812 

### Type of change

- Bug fix

### How risky is the change?

- Not relevant

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

Commit:
RavenDB-17812 Making sure we wait a bit before getting index performance. Otherwise we will have a race in some of the exestuations.

arekpalinski commented on test/SlowTests/Issues/RavenDB_3417.cs:
```diff
@@ -27,16 +27,25 @@ public void GetIndexingPerformanceStatisticsShouldWork()
+
+                    return true;
+                    
+                }, true, 5000));
```
Previously we had 3 separate assertions. Now we aren't able to distinguish between them. We'll just get `Assert.True()` failure

omerlewitz replied:
suggestion?

omerlewitz replied:
correct

arekpalinski commented on test/SlowTests/Issues/RavenDB_3417.cs:
```diff
@@ -27,16 +27,25 @@ public void GetIndexingPerformanceStatisticsShouldWork()
                 WaitForIndexing(store);
 
-                var indexingPerformanceStatistics = store.Maintenance.Send(new GetIndexPerformanceStatisticsOperation());
+                Assert.True(WaitForValue(() =>
```
What is the root cause that we need to repeat the operation? Do we need that in other tests as well?

maciejaszyk replied:
Maybe we should introduce a `WaitForIndexPerformanceStatistics` and fix other tests too?

omerlewitz replied:
There's only one more test that uses it.
There's no need to wait over there because we already wait as far as I see.

maciejaszyk commented:
I think this should be applied to 4.2 version or there is something new that caused this in 5.3?

Commit:
RavenDB-17812 Resolved conversation

The reason of using string instead of throwing exception is that we want to try run it for few times other than one time and throwing exception.
